activemodel/test/cases/errors_test.rb

```diff
@@ -189,6 +189,33 @@ def test_has_no_key
     assert !person.errors.added?(:name, "cannot be blank")
   end
 
+  test "add_once adds a specific message to an attribute only once" do
+    person = Person.new
+    message = "has only one error message"
+    person.errors.add_once(:name, message)
+    person.errors.add_once(:name, message)
+    assert_equal [message], person.errors[:name]
+    assert_not_equal [message, message], person.errors[:name]
```

Comment:
I think i was just being super explicit in that it should only have that message once. It is removable but I don't think the double assert is too restrictive or confusing. I actually think its clearer than a single assert when it comes to the spec of ```add_once```. However if it should be removed it will assert the same thing.